lib/cli/file-set-pipeline/traverse.js

```diff
@@ -27,14 +27,15 @@ var Traverser = require('../traverser');
  * @param {CLI} cli
  */
 function traverse(cli) {
-    var traverser = new Traverser({
-        'extensions': cli.extensions,
-        'ignorePath': cli.ignorePath,
-        'detectIgnore': cli.detectIgnore,
-        'cwd': cli.cwd
-    });
-
-    cli.files = traverser.traverse(cli.globs);
+    if (!cli.files) {
+        cli.traverser = new Traverser({
+            'extensions': cli.extensions,
+            'ignorePath': cli.ignorePath,
+            'detectIgnore': cli.detectIgnore,
+            'cwd': cli.cwd
+        });
+    }
+    cli.files = cli.traverser.traverse(cli.globs);
```

Comment:
@wooorm Check this out. We have to cache the traverser, so I went with the `cli.traverser` approach.